moderntribe/the-events-calendar
# Resolution for Grid Stacking on V2

🎟 [TEC-3208] [TEC-3209] [TEC-3212] [TEC-3210]
---
On these four bugs, we found that our All Day, Single Day and Multi-day Events calculations for Grids (week/month) was broken by permutations on End of day Cut-off, Timezones, Timezone Mode and Start of Week settings.

Our goal with this PR is to solve all 4 tickets related to data querying.

We made specific modifications to:
- How we determine which event needs to be displayed on a week by week basis with some caching.
- Determinations for Multiday events now will have more case scenarios 
- Events now are limited to spawning only 7 days per week, which was breaking for long spawning events with Timezones.
- Proper data fetching for events from the database when using Timezone Mode for Site-wide

We are specifically not giving screencasts here, but we passed proper guidance to the parameters of tests so we can cover as much ground as possible.

#### Testing Instructions

For Month and Week views, verify the following settings work correctly:

1. Week Starts On
2. Timezone (WP)
3. Timezone Mode (WP vs Event)
4. Modifying event timezone with both timezone modes
5. End of Day Cutoff
6. Month view events per day
7. events per page (list-style views)

We need to test All Day, Multi-day, and regular/standard events.

For all other views, just run a simple pass.
Including, Events per page for list-style views.

[TEC-3208]: https://moderntribe.atlassian.net/browse/TEC-3208
[TEC-3209]: https://moderntribe.atlassian.net/browse/TEC-3209
[TEC-3212]: https://moderntribe.atlassian.net/browse/TEC-3212
[TEC-3210]: https://moderntribe.atlassian.net/browse/TEC-3210
